### PR TITLE
Fixed unit tests

### PR DESCRIPTION
**Pull request by VysotskiVadim**
This PR contains fixed for unit tests which were broken by a new mapbox java and found in #5751

### CODE REVIEW

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/utils/DirectionsResponseUtils.kt [hunk 1]
```diff
@@ -33,7 +34,7 @@ suspend fun parseDirectionsResponse(
         } catch (ex: Exception) {
             when (ex) {
                 is JSONException,
-                is NullPointerException -> ExpectedFactory.createError(ex)
+                is JsonSyntaxException -> ExpectedFactory.createError(ex)
```
Comment (VysotskiVadim) [inline libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/utils/DirectionsResponseUtils.kt:37]: Now mapbox-java throws a different error:

image (3)

Comment (VysotskiVadim) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/utils/DirectionsResponseUtils.kt:37]: TBH I'm not sure if it's an issue of mapbox-java trowing different exception or us relying on null pointer exception

File: libnavigation-router/src/test/java/com/mapbox/navigation/route/internal/RouterWrapperTests.kt [hunk 2]
```diff
@@ -132,17 +132,9 @@ class RouterWrapperTests {
...
-        every { value } returns "{\"message\":\"should be >= 1\",\"code\":\"InvalidInput\"}"
-        every { error } returns null
+    private val routerResultSuccessErroneousValue: Expected<RouterError, String>
+        = ExpectedFactory.createValue("{\"message\":\"should be >= 1\",\"code\":\"InvalidInput\"}")
 
-        val valueSlot = slot<Expected.Transformer<String, Unit>>()
-        every { fold(any(), capture(valueSlot)) } answers {
-            valueSlot.captured.invoke(this@mockk.value!!)
-        }
-    }
```
Comment (VysotskiVadim) [inline libnavigation-router/src/test/java/com/mapbox/navigation/route/internal/RouterWrapperTests.kt:145]: I don't understand profit of mocking here 🤔

File: libnavigation-router/src/test/java/com/mapbox/navigation/route/internal/RouterWrapperTests.kt [hunk 3]
```diff
@@ -368,10 +360,13 @@ class RouterWrapperTests {
 
     @Test
     fun `route refresh fails with null requestUuid`() {
-        val route: DirectionsRoute = mockk(relaxed = true)
```
Comment (VysotskiVadim) [inline libnavigation-router/src/test/java/com/mapbox/navigation/route/internal/RouterWrapperTests.kt:371]: mockk failed with an error `ailed to access class com.mapbox.api.directions.v5.models.DirectionsJsonObject$Builder from class com.mapbox.api.directions.v5.models.DirectionsRoute$Builder$Subclass28 (com.mapbox.api.directions.v5.models.DirectionsJsonObject$Builder is in unnamed module of loader org.robolectric.internal.AndroidSandbox$SdkSandboxClassLoader @58601e7a; com.mapbox.api.directions.v5.models.DirectionsRoute$Builder$Subclass28 is in unnamed module of loader net.bytebuddy.dynamic.loading.MultipleParentClassLoader @6a314f75)`

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/utils/DirectionsResponseUtils.kt (hunk 1 above)
Comment (VysotskiVadim) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/utils/DirectionsResponseUtils.kt:37]: The error type doesn't seem like part of mapbox-java contract. I think we just shouldn't rely on the error type or we need to be ready to update it when mapbox-java changes it.
@mapbox/navigation-android , WDYT?

**Review by Guardiola31337 — APPROVED** [review]
Thanks for the quick fix 🙇‍♂️

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/utils/DirectionsResponseUtils.kt (hunk 1 above)
Comment (LukasPaczos) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/utils/DirectionsResponseUtils.kt:37]: I'm wondering whether we should just catch all exception here and deliver them back to developers as errors.

File: libnavigation-router/src/test/java/com/mapbox/navigation/route/internal/RouterWrapperTests.kt (hunk 3 above)
Comment (LukasPaczos) [reply libnavigation-router/src/test/java/com/mapbox/navigation/route/internal/RouterWrapperTests.kt:371]: This is surprising, do we know why?

**Comment by RingerJK** [comment]
interesting bug, looks like mockk limitations

File: libnavigation-router/src/test/java/com/mapbox/navigation/route/internal/RouterWrapperTests.kt (hunk 3 above)
Comment (LukasPaczos) [reply libnavigation-router/src/test/java/com/mapbox/navigation/route/internal/RouterWrapperTests.kt:371]: I would imagine developers mocking the route as well.

Comment (Guardiola31337) [reply libnavigation-router/src/test/java/com/mapbox/navigation/route/internal/RouterWrapperTests.kt:371]: Yeah, that's a great point @VysotskiVadim

Comment (RingerJK) [reply libnavigation-router/src/test/java/com/mapbox/navigation/route/internal/RouterWrapperTests.kt:371]: what I found https://github.com/mockk/mockk/issues/454 https://github.com/mockk/mockk/issues/724 and so on. Looks like we need to update mockk

**Pull request merged**